gateway/src/resty/resolver.lua

```diff
@@ -131,10 +124,23 @@ function _M.parse_nameservers(path)
     -- see https://github.com/3scale/apicast/issues/321 for more details
     insert(nameservers, resolver)
   end
+  
+  for _,line in ipairs(re.split(resolv_conf, "\n+")) do
+    
+    local domains = match(line, '^search%s+([^\n]+)')
+    
+    if domains then
+      ngx.log(ngx.DEBUG, 'search ', domains)
+
+      for domain in gmatch(domains or '', '([^%s]+)') do
+        ngx.log(ngx.DEBUG, 'search domain: ', domain)
+        insert(search, domain)
+      end
+    end
 
-  for server in gmatch(resolv_conf, 'nameserver%s+([^%s]+)') do
+    local server = match(line, '^nameserver%s+([^%s]+)')
     -- TODO: implement port matching based on https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=549190
-    if server ~= resolver then
+    if server and server ~= resolver then
```

Comment:
I think we don't have tests for this case.

Reply:
Actual this test case fails.

We have agreed (me and @davidor)  that we should address those problems in another PR.

The issue which treat this case is #630